radixdlt/src/main/java/org/radix/utils/Range.java

```diff
@@ -54,14 +54,18 @@ protected void setHigh(T high)
 	}
 
 	@Override
-	public boolean equals(Object other)
+	public final boolean equals(Object other)
 	{
 		if (other == this)
 			return true;
 
 		if (other == null)
```

Comment:
Not your code, but `null` is never an `instanceof` anything, so this check is redundant.

Reply:
removed `if (other == null); return false`

Reply:
@clstrfsck Spoke to soon, actually this seems needed, at least I did not get it to work by changing this:

```java
		if (other == null)
			return false;

		// `FooRange` should not automatically equal `(Range) BarRange`.
		if (!this.getClass().equals(other.getClass()))
			return false;
```

into just this:

```java
		// `FooRange` should not automatically equal `(Range) BarRange`.
		if (!Objects.equals(this.getClass(), other.getClass()))
			return false;
```